### ISSUE TITLE
Gossip Encryption autogenerate fails in Openshift on K8s 1.24 when using PSPs

### ISSUE THREAD

**Issue by adaumann** [hashicorp/consul-k8s#1503]
If using Openshift with Consul/helm, The 

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request. Searching for pre-existing feature requests helps us consolidate datapoints for identical requirements into a single place, thank you!
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

---

### Overview of the Issue

Currently using Consul Helm within Openshift 4.11.1 context (with Kubernetes 1.24). The Gossip Encryption fails, even if the openshift.enabled flag is set, with following error:

Error creating: pods "consul-gossip-encryption-autogenerate-" is forbidden: unable to validate against any security context constraint: [provider "anyuid": Forbidden: not usable by user or serviceaccount, provider restricted-v2: .spec.securityContext.fsGroup: Invalid value: []int64{1000}: 1000 is not an allowed group, spec.containers[0].securityContext.runAsUser: Invalid value: 100: must be in the ranges: [1000660000, 1000669999], provider "restricted": Forbidden: not usable by user or serviceaccount, provider "nonroot-v2": Forbidden: not usable by user or serviceaccount, provider "nonroot": Forbidden: not usable by user or serviceaccount, provider "hostmount-anyuid": Forbidden: not usable by user or serviceaccount, provider "machine-api-termination-handler": Forbidden: not usable by user or serviceaccount, provider "hostnetwork-v2": Forbidden: not usable by user or serviceaccount, provider "hostnetwork": Forbidden: not usable by user or serviceaccount, provider "hostaccess": Forbidden: not usable by user or serviceaccount, provider "privileged": Forbidden: not usable by user or serviceaccount]

My expectation is that with the flag openshift.enabled all security contexts are set corrently. Not only for the server, client etc. Also for the init containers. Additionally took the latest images from redhat, see below.

### Reproduction Steps

Steps to reproduce this issue, eg:

1. When running helm install with the following `values.yml`:
```
consul:
  global:
    name: consul
    datacenter: dc1
    image: registry.connect.redhat.com/hashicorp/consul:1.13.1-ubi
    imageK8S: registry.connect.redhat.com/hashicorp/consul-k8s-control-plane:0.48.0-ubi
    imagePullSecrets:
      - name: 15472453-hcv-redhat-registry-pull-secret
    openshift:
      enabled: true
    gossipEncryption:
      autoGenerate: true
    tls:
      enabled: true
      enableAutoEncrypt: true
      verify: false
      serverAdditionalDNSSANs:
        - localhost
        - consul-server
        - consul-server.sbk-hcv-infra
        - consul-server.sbk-hcv-infra.svc.cluster.local     
    acls:
      manageSystemACLs: true 
```
1. View error

Error creating: pods "consul-gossip-encryption-autogenerate-" is forbidden: unable to validate against any security context constraint: [provider "anyuid": Forbidden: not usable by user or serviceaccount, provider restricted-v2: .spec.securityContext.fsGroup: Invalid value: []int64{1000}: 1000 is not an allowed group, spec.containers[0].securityContext.runAsUser: Invalid value: 100: must be in the ranges: [1000660000, 1000669999], provider "restricted": Forbidden: not usable by user or serviceaccount, provider "nonroot-v2": Forbidden: not usable by user or serviceaccount, provider "nonroot": Forbidden: not usable by user or serviceaccount, provider "hostmount-anyuid": Forbidden: not usable by user or serviceaccount, provider "machine-api-termination-handler": Forbidden: not usable by user or serviceaccount, provider "hostnetwork-v2": Forbidden: not usable by user or serviceaccount, provider "hostnetwork": Forbidden: not usable by user or serviceaccount, provider "hostaccess": Forbidden: not usable by user or serviceaccount, provider "privileged": Forbidden: not usable by user or serviceaccount]
### Expected behavior

Security context is set correctly.

**Comment by adaumann** [comment]
Similar error for TLS-Init:

W0915 14:01:14.535687  450508 warnings.go:70] would violate PodSecurity "restricted:v1.24": allowPrivilegeEscalation != false (container "tls-init" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "tls-init" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "tls-init" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "tls-init" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")

**Comment by david-yu** [comment]
Thanks appreciate the heads up, it sounds like maybe these issues are isolated to OpenShift 4.11 (K8s 1.24)?

**Comment by kschoche** [comment]
I was able to reproduce this using Pod Security Standards on Kind.
@adaumann is this enabled by default in OpenShift 4.11 or something you've set manually on your cluster?
 
```
demo $ kubectl label --overwrite ns consul \
   pod-security.kubernetes.io/enforce=restricted \
   pod-security.kubernetes.io/enforce-version=v1.24
namespace/consul labeled
demo $
demo $ helm install consul hashicorp/consul -f val.yaml -n consul

[ tls-init fails in the same way ]
```

**Comment by adaumann** [comment]
Hello,
Openshift I think abstracts namespaces by "projects". The namespaces are hidden mostly. A project is a security enhanced namespace in my view, as example:

````
kind: Project
apiVersion: project.openshift.io/v1
metadata:
  name: rabbitmq
  uid: 6dd84cae-e6de-46a8-b012-dad05c025b11
  resourceVersion: '32731'
  creationTimestamp: '2022-09-20T14:11:46Z'
  labels:
    kubernetes.io/metadata.name: rabbitmq
    pod-security.kubernetes.io/audit: restricted
    pod-security.kubernetes.io/audit-version: v1.24
    pod-security.kubernetes.io/warn: restricted
    pod-security.kubernetes.io/warn-version: v1.24
  annotations:
    openshift.io/description: ''
    openshift.io/display-name: ''
    openshift.io/requester: kubeadmin
    openshift.io/sa.scc.mcs: 's0:c26,c0'
    openshift.io/sa.scc.supplemental-groups: 1000650000/10000
    openshift.io/sa.scc.uid-range: 1000650000/10000
  managedFields:
    - manager: Go-http-client
      operation: Update
      apiVersion: v1
      time: '2022-09-20T14:11:46Z'
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:annotations':
            .: {}
            'f:openshift.io/description': {}
            'f:openshift.io/display-name': {}
            'f:openshift.io/requester': {}
          'f:labels':
            .: {}
            'f:kubernetes.io/metadata.name': {}
    - manager: cluster-policy-controller
      operation: Update
      apiVersion: v1
      time: '2022-09-20T14:11:46Z'
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:annotations':
            'f:openshift.io/sa.scc.mcs': {}
            'f:openshift.io/sa.scc.supplemental-groups': {}
            'f:openshift.io/sa.scc.uid-range': {}
          'f:labels':
            'f:pod-security.kubernetes.io/audit': {}
            'f:pod-security.kubernetes.io/audit-version': {}
            'f:pod-security.kubernetes.io/warn': {}
            'f:pod-security.kubernetes.io/warn-version': {}
    - manager: openshift-controller-manager
      operation: Update
      apiVersion: v1
      time: '2022-09-20T14:11:46Z'
      fieldsType: FieldsV1
      fieldsV1:
        'f:spec':
          'f:finalizers': {}
      subresource: finalize
spec:
  finalizers:
    - kubernetes
status:
  phase: Active
````

**Comment by david-yu** [comment]
Closing as this should now be addressed via Consul K8s 1.2.1, 1.1.4, and 1.0.9.
